csrc/multidevice/utils.h

```diff
@@ -127,4 +129,44 @@ int64_t getShardedAxis(TensorView*);
 
 // Reorders a TensorView so that the DID parallelized axis are in front.
 void reorderDIDToFront(TensorView*);
+
+// Given a TensorView and the shape of a sharded tensor of which certain
+// dimensions are partially alloated, returns the global shape that'll be used
+// to bind to the TensorView's logical domain. This is to solve #3282 so we can
+// bind a sharded tensor to a TensorView that has a DID-parallel loop domain.
+//
+// For example, when `tv` is
+//   logical: iM, iN
+//   allocation: iDIDx{D}, iN/D, iM
+// and `sizes` is [2, 3], the returned shape will be [2, 3D]. This is because,
+// according to the allocation domain, iM is fully allocated and iN is sharded
+// and thus partially allocated.
+//
+// As a degenerate case, it's fine to call this function with a non-sharded
+// TensorView and tensor.
+//
+// Limitations:
+// - The function assumes that there are no Merges from logical to the
+// DID-parallel IterDomains in allocation. Otherwise, it's unclear which logical
```

Comment:
For my own understanding, can you give an example where this is unclear?

Reply:
```
t = makeContigTensor(2);
t->merge(0, 1);
t->axis(0)->parallelize(DIDx);
```

Given a runtime tensor of local shape [1, 1], it's unclear the global shape is [1, D] or [D, 1] or even [2, D/2], etc.

Reply:
Ok, I see. The reason is that basically if we only know the extent of `i0 * i1` we cannot deduce `i0` and `i1` in a unique way. So I agree it is reasonable to simply rule out this case.